connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/decoder/ActionDecoder.java

```diff
@@ -0,0 +1,122 @@
+package com.redhat.cloud.notifications.connector.email.decoder;
+
+import com.redhat.cloud.notifications.ingress.ActionOut;
+import com.redhat.cloud.notifications.ingress.Application;
+import com.redhat.cloud.notifications.ingress.Bundle;
+import com.redhat.cloud.notifications.ingress.Context;
+import com.redhat.cloud.notifications.ingress.Event;
+import com.redhat.cloud.notifications.ingress.EventType;
+import com.redhat.cloud.notifications.ingress.Metadata;
+import com.redhat.cloud.notifications.ingress.Payload;
+import com.redhat.cloud.notifications.ingress.Source;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+public class ActionDecoder {
+
+    private ActionDecoder() { }
+
+    /**
+     * Decodes the incoming raw action from the engine to an object.
+     * @param rawAction the incoming raw action {@link String}.
+     * @return a populated {@link ActionOut} pojo.
+     */
+    public static ActionOut decode(final String rawAction) {
```

Comment:
Is this class different from `Parser`?
https://github.com/RedHatInsights/notifications-backend/blob/3b34486d027298d253dd146f3d99021529182609/engine/src/main/java/com/redhat/cloud/notifications/utils/ActionParser.java#L13